### PR TITLE
Feature/83 request headers

### PR DESCRIPTION
**Pull request by sjongejan**

### CODE REVIEW

File: app/Http/Controllers/Api/RequestApiController.php [hunk 1]
```diff
@@ -42,23 +44,31 @@ public function showSingleRequest(int $requestId): JsonResponse
...
-        $endpoint = \App\Endpoint::firstOrCreate($request->only('url', 'method'));
-        $endpointRequest = $endpoint->requests()->create();
+        /* @var Endpoint $endpoint */
+        $endpoint = \App\Endpoint::firstOrCreate($webRequest->only('url', 'method'));
 
-        $this->dispatch(new ExecuteRequestJob($endpointRequest));
+        /* @var Request $request */
+        $request = $endpoint->requests()->create();
+
+        $request->requestHeaders()->createMany($webRequest->get('request_headers', []));
```
Comment (FaaPz) [inline app/Http/Controllers/Api/RequestApiController.php]: Empty `[]`?

File: app/Http/Controllers/Api/RequestApiController.php [hunk 2]
```diff
@@ -42,23 +44,31 @@ public function showSingleRequest(int $requestId): JsonResponse
      * Create a new request for the endpoint.
      * Dispatch ExecuteRequestJob.
      *
-     * @param Request $request
+     * @param WebRequest $webRequest
```
Comment (FaaPz) [inline app/Http/Controllers/Api/RequestApiController.php]: I don't like this rename... but I also don't have a better one. It's good for now. 😬

File: app/Http/Controllers/Api/RequestApiController.php [hunk 3]
```diff
@@ -42,23 +44,31 @@ public function showSingleRequest(int $requestId): JsonResponse
...
      */
-    public function storeRequest(Request $request): Response
+    public function storeRequest(WebRequest $webRequest): Response
     {
-        $this->validate($request, [
+        $this->validate($webRequest, [
             'url' => 'required|url',
             'method' => 'required|in:'.implode(',', \App\Request::getAllowedMethods()),
+            'request_headers' => 'array',
+            'request_headers.*.name' => 'required',
```
Comment (FaaPz) [inline app/Http/Controllers/Api/RequestApiController.php]: `required|string|max:255` 👈

File: app/Http/Controllers/Api/RequestApiController.php [hunk 4]
```diff
@@ -42,23 +44,31 @@ public function showSingleRequest(int $requestId): JsonResponse
...
+            'request_headers.*.value' => 'required',
```
Comment (FaaPz) [inline app/Http/Controllers/Api/RequestApiController.php]: `required|string|max:16777215` 👈

Comment (FaaPz) [reply app/Http/Controllers/Api/RequestApiController.php]: Maybe is the actual max. size of `MEDIUMTEXT` to much... 🤔

**Review by FaaPz — CHANGES_REQUESTED** [review]
Minor feedback. Good work! 👍

File: app/Http/Controllers/Api/RequestApiController.php (hunk 1 above)
Comment (sjongejan) [reply app/Http/Controllers/Api/RequestApiController.php]: Empty [] is the default value for request_headers when not present in the request.

**Review by FaaPz — APPROVED** [review]

**Pull request merged**